### PR TITLE
Create check-in on reading log add

### PR DESCRIPTION
**Pull request by jimchamp**
Automatically creates a check-in event whenever a book is added to "Currently Reading" or "Already Read" reading logs.

Blocked by #6662 --- Once that is merged, this will need to be rebased.

### CODE REVIEW

File: openlibrary/core/bookshelves_events.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+from . import db
+
+
+class BookshelfEvent(IntEnum):
+    START = 1
+    UPDATE = 2
+    FINISH = 3
+
+    @classmethod
+    def has_key(cls, key: str) -> bool:
```
Comment (cclauss) [inline openlibrary/core/bookshelves_events.py:12]: Will this work?  `has_key()` is kinda passe.
```suggestion
    def __contains__(cls, key: str) -> bool:
```
https://portingguide.readthedocs.io/en/latest/dicts.html

File: openlibrary/plugins/openlibrary/api.py [hunk 2]
```diff
@@ -329,6 +331,23 @@ def POST(self, work_id):
 
         else:
             edition_id = int(i.edition_id.split('/')[2][2:-1]) if i.edition_id else None
+
+            # Create check-in if "Currently Reading" or "Already Read" is clicked
+            if edition_id and bookshelf_id in (2, 3):
+                today = date.today()
+                date_str = f'{today.year}-{today.month:02}-{today.day:02}'
```
Comment (cclauss) [inline openlibrary/plugins/openlibrary/api.py:338]:
```suggestion
                date_str = f"{date.today():%Y-%m-%d}"
```

File: openlibrary/plugins/upstream/checkins.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            return web.badrequest(message="Invalid request")
+        return delegate.RawText(json.dumps({'status': 'ok'}))
+
+    def is_valid(self, data: dict) -> bool:
+        """Validates POSTed check-in data."""
+        if not all(key in data for key in ('edition_olid', 'year', 'event_type')):
+            return False
+        if not BookshelfEvent.has_key(data['event_type']):
+            return False
+        return True
```
Comment (cclauss) [inline openlibrary/plugins/upstream/checkins.py:74]:
```suggestion
        return BookshelfEvent.has_key(data['event_type'])
```

File: openlibrary/core/bookshelves_events.py (hunk 1 above)
Comment (jimchamp) [reply openlibrary/core/bookshelves_events.py:12]: Thanks for this note!  I tried this and was seeing `TypeDef` errors when checking if an int was in `BookshelfEvents`.  I applied this review note and fixed the error in #7212

**Comment by jimchamp** [comment]
Closing this as we have decided that patrons should check in manually.

**Pull request closed without merging**